File: Example ObjC/Example ObjC/PlayerViewController.m
//
//  PlayerViewController.m
//  FlowplayerExample-ObjectiveC
//
//  Created by Mehdi on 5. 7. 2021..
//

#import "PlayerViewController.h"

@interface PlayerViewController () {}
@end

@implementation PlayerViewController

- (void)viewDidAppear:(BOOL)animated {
	_flowplayerViewController = [[FPFlowplayerViewController alloc] init];
	FPPlayerControlConfigBuilder * builder = [[FPPlayerControlConfig Builder] setMuteControl:true];
	
	builder = [builder enablePlugins:_plugins];
	builder = [builder setCustomWithKey:@"speed.options" value:@[@0.5, @1, @2, @5]];
	builder = [builder setCustomWithKey:@"speed.labels" value:@[@"Slow", @"Normal", @"Double", @"Fast"]];

	[_flowplayerViewController setControlConfig: [builder build]];

	[_containerView addSubview:_flowplayerViewController.view];

	// Add the delegate after the flowplayerViewController has been added to the container.
	[_flowplayerViewController addPlayerDelegate:self];

	if([self flowplayerMedia]) {
		[_flowplayerViewController prepareWithFlowplayerMedia:[self flowplayerMedia] autoStart:true];
	} else if ([self externalMedia]) {
		[_flowplayerViewController prepareWithExternalMedia:[self externalMedia] autoStart:true];
	}
}

- (void)onAdBreakCompleteWithEvent:(FPAdBreakCompleteEvent * _Nonnull)event {
	printf("OnAdBreakComplete");
}

- (void)onAdBreakStartWithEvent:(FPAdBreakStartEvent * _Nonnull)event {
	printf("OnAdBreakStart");
}

- (void)onAdClickWithEvent:(FPAdClickEvent * _Nonnull)event {
	printf("OnAdClick");
}

- (void)onAdCompleteWithEvent:(FPAdCompleteEvent * _Nonnull)event {
	printf("OnAdComplete");
}

- (void)onAdErrorWithEvent:(FPAdErrorEvent * _Nonnull)event {
	printf("OnAdError");
}

- (void)onAdPauseWithEvent:(FPAdPauseEvent * _Nonnull)event {
	printf("OnAdPause");
}

- (void)onAdResumeWithEvent:(FPAdResumeEvent * _Nonnull)event {
	printf("OnAdResume");
}

- (void)onAdSkipWithEvent:(FPAdSkipEvent * _Nonnull)event {
	printf("OnAdSkip");
}

- (void)onAdStartWithEvent:(FPAdStartEvent * _Nonnull)event {
	printf("OnAdStart");
}

- (void)onBufferWithEvent:(FPBufferEvent * _Nonnull)event {
	printf("OnBuffer");
}

- (void)onCompleteWithEvent:(FPCompleteEvent * _Nonnull)event {
	printf("OnComplete");
}

- (void)onErrorWithEvent:(FPErrorEvent * _Nonnull)event {
	printf("OnError");
}

- (void)onFullscreenWithEvent:(FPFullscreenEvent * _Nonnull)event {
	printf("OnFullscreen");
}

- (void)onMuteWithEvent:(FPMuteEvent * _Nonnull)event {
	printf("OnMuteChanged");
}

- (void)onVolumeWithEvent:(FPVolumeEvent * _Nonnull)event {
	printf("OnVolumeChanged");
}

- (void)onIdleWithEvent:(FPIdleEvent * _Nonnull)event {
	printf("OnIdle");
}

- (void)onPauseWithEvent:(FPPauseEvent * _Nonnull)event {
	printf("OnPause");
}

- (void)onPlayWithEvent:(FPPlayEvent * _Nonnull)event {
	printf("OnPlay");
}

- (void)onSpeedWithEvent:(FPSpeedEvent * _Nonnull)event {
	printf("OnSpeed");
}

- (void)onAudioTracksWithEvent:(FPAudioTracksEvent *)event {
	printf("OnAudioTracks");
}

- (void)onAudioTrackSelectWithEvent:(FPAudioTrackSelectEvent *)event {
	printf("OnAudioTrackSelect");
}

- (void)onOvpMetadataWithEvent:(FPOvpMetadataEvent * _Nonnull)event {
	printf("onOvpMetadataWithEvent");
}

- (void)onCastingWithEvent:(FPCastingEvent * _Nonnull)event {
	printf("onCastingWithEvent");
}

- (void)onSubtitleTrackSelectWithEvent:(FPSubtitleTrackSelectEvent * _Nonnull)event {
	printf("FPSubtitleTrackSelectEvent");
}

- (void)onSubtitleTracksWithEvent:(FPSubtitleTracksEvent * _Nonnull)event {
	printf("FPSubtitleTracksEvent");
}

- (void)onReadyWithEvent:(FPReadyEvent * _Nonnull)event {
	printf("onReadyEvent");
}

- (void)encodeWithCoder:(nonnull NSCoder *)coder {
	printf("encodeWithCoder");
}

- (void)traitCollectionDidChange:(nullable UITraitCollection *)previousTraitCollection {
	printf("traitCollectionDidChange");
}

- (void)preferredContentSizeDidChangeForChildContentContainer:(nonnull id<UIContentContainer>)container {
	printf("preferredContentSizeDidChangeForChildContentContainer");
}

- (CGSize)sizeForChildContentContainer:(nonnull id<UIContentContainer>)container withParentContainerSize:(CGSize)parentSize {
	printf("sizeForChildContentContainer");
	return CGSizeZero;
}

- (void)systemLayoutFittingSizeDidChangeForChildContentContainer:(nonnull id<UIContentContainer>)container {
	printf("systemLayoutFittingSizeDidChangeForChildContentContainer");
}

- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(nonnull id<UIViewControllerTransitionCoordinator>)coordinator {
	printf("viewWillTransitionToSize");
}

- (void)willTransitionToTraitCollection:(nonnull UITraitCollection *)newCollection withTransitionCoordinator:(nonnull id<UIViewControllerTransitionCoordinator>)coordinator {
	printf("willTransitionToTraitCollection");
}

- (void)didUpdateFocusInContext:(nonnull UIFocusUpdateContext *)context withAnimationCoordinator:(nonnull UIFocusAnimationCoordinator *)coordinator {
	printf("didUpdateFocusInContext");
}

- (void)setNeedsFocusUpdate {
	printf("setNeedsFocusUpdate");
}

- (BOOL)shouldUpdateFocusInContext:(nonnull UIFocusUpdateContext *)context {
	return false;
}

- (void)updateFocusIfNeeded {
	printf("shouldUpdateFocusInContext");
}

@end

File: Example Swift/Example Swift/PlayerViewController.swift
//  PlayerViewController.swift
//  Example Swift
//
//  Created by Mehdi on 9. 2. 2022.

import UIKit
import Flowplayer

class PlayerController: UIViewController  {
	
	// - IBOutlets
	@IBOutlet weak var containerView: UIView!
	@IBOutlet weak var mediaLabel: UILabel!
	
	// - Views
	private lazy var videoController = FPFlowplayerViewController()
	
	// - Props
	var mediaName: String?
	var media: FPMedia?
	var plugins: [String]?
	var useCustomWebControls = false
	
	// - Lifecycle
	override func viewWillAppear(_ animated: Bool) {
		super.viewWillAppear(animated)
		mediaLabel.text = mediaName
	}
	
	override func viewDidAppear(_ animated: Bool) {
		super.viewDidAppear(animated)

		// Create and add config
		let config = createConfig()
		videoController.setControlConfig(config)
		
		// - Add video player to view
		containerView.addSubview(videoController.view)

		// Add the delegate after the flowplayerViewController has been added to the container.
		videoController.addPlayerDelegate(self)
		
		// Prepare media to play
		if let flowplayerMedia = media as? FPFlowplayerMedia {
			videoController.prepare(flowplayerMedia: flowplayerMedia, autoStart: true)
		}
		
		if let externalMedia = media as? FPExternalMedia {
			videoController.prepare(externalMedia: externalMedia, autoStart: true)
		}
	}
}

// MARK: - Private
extension PlayerController {
	
	private func createConfig() -> FPPlayerControlConfig {
		let builder =  FPPlayerControlConfigBuilder()
		builder.setMuteControl(true)
		builder.enablePlugins(plugins!)
		
		if (plugins!.contains("speed")) {
			builder.setCustom(key: "speed.options", value: [0.5, 1, 2, 5])
			builder.setCustom(key: "speed.labels", value: ["Slow", "Normal", "Double", "Fast"])
		}
		
		if (useCustomWebControls) {
			builder.setControlsUrl("https://fp-eu-w1-nnarhinen.s3-eu-west-1.amazonaws.com/sdk-controls/custom.html?token=[TOKEN]&[CONFIG]")
		}
		
		return builder.build()
	}
	
}


// MARK: - FPFlowplayerDelegate
extension PlayerController: FPFlowplayerDelegate {
	
	func onIdle(event: FPIdleEvent) {
		print("onIdle")
	}
	
	func onReady(event: FPReadyEvent) {
		print("onReady")
	}
	
	func onPlay(event: FPPlayEvent) {
		print("onPlay")
	}
	
	func onPause(event: FPPauseEvent) {
		print("onPause")
	}
	
	func onBuffer(event: FPBufferEvent) {
		print("onBuffer")
	}
	
	func onComplete(event: FPCompleteEvent) {
		print("onComplete")
	}
	
	func onFullscreen(event: FPFullscreenEvent) {
		print("onFullscreen")
	}
	
	func onMute(event: FPMuteEvent) {
		print("onMute")
	}
	
	func onVolume(event: FPVolumeEvent) {
		print("onVolume")
	}
	
	func onSpeed(event: FPSpeedEvent) {
		print("onSpeed")
	}
	
	func onAudioTracks(event: FPAudioTracksEvent) {
		print("onAudioTracks")
	}
	
	func onOvpMetadata(event: FPOvpMetadataEvent) {
		print("onOvpMetadata")
	}
	
	func onAudioTrackSelect(event: FPAudioTrackSelectEvent) {
		print("onAudioTrackSelect")
	}
	
	func onError(event: FPErrorEvent) {
		print("onError")
	}
	
	func onAdBreakStart(event: FPAdBreakStartEvent) {
		print("onAdBreakStart")
	}
	
	func onAdBreakComplete(event: FPAdBreakCompleteEvent) {
		print("onAdBreakComplete")
	}
	
	func onAdStart(event: FPAdStartEvent) {
		print("onAdStart")
	}
	
	func onAdPause(event: FPAdPauseEvent) {
		print("onAdPause")
	}
	
	func onAdResume(event: FPAdResumeEvent) {
		print("onAdResume")
	}
	
	func onAdClick(event: FPAdClickEvent) {
		print("onAdClick")
	}
	
	func onAdSkip(event: FPAdSkipEvent) {
		print("onAdSkip")
	}
	
	func onAdComplete(event: FPAdCompleteEvent) {
		print("onAdComplete")
	}
	
	func onAdError(event: FPAdErrorEvent) {
		print("onAdError")
	}
	
	func onSubtitleTrackSelect(event: FPSubtitleTrackSelectEvent) {
		print("onSubtitleTrackSelect")
	}
	
	func onSubtitleTracks(event: FPSubtitleTracksEvent) {
		print("onSubtitleTracks")
	}
	
	func onCasting(event: FPCastingEvent) {
		print("onCasting")
	}

}
